pkgs/desktops/gnome-2/platform/gtkhtml/4.x.nix

```diff
@@ -1,16 +1,23 @@
-{ stdenv, fetchurl, pkg-config, gtk3, intltool
+{ stdenv, fetchFromGitLab, pkg-config, gtk3, intltool, autoreconfHook
 , GConf, enchant, isocodes, gnome-icon-theme, gsettings-desktop-schemas }:
 
 stdenv.mkDerivation rec {
   version = "4.10.0";
   pname = "gtkhtml";
 
-  src = fetchurl {
-    url = "mirror://gnome/sources/gtkhtml/4.10/${pname}-${version}.tar.xz";
-    sha256 = "1hq6asgb5n9q3ryx2vngr4jyi8lg65lzpnlgrgcwayiczcj68fya";
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "Archive";
+    repo = "gtkhtml";
+    rev = "master";
+    sha256 = "sha256-jL8YADvhW0o6I/2Uo5FNARMAnSbvtmFp+zWH1yCVvQk=";
   };
-
   propagatedBuildInputs = [ gsettings-desktop-schemas gtk3 gnome-icon-theme GConf ];
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ intltool enchant isocodes ];
+  buildInputs = [ intltool enchant isocodes autoreconfHook ];
+
+  patches = [
+    # Same patch as in AUR, but different format: https://aur.archlinux.org/cgit/aur.git/tree/enchant-2.patch?h=gtkhtml4-git
```

Comment:
Many questions I will arise here.

1. If this is "the same patch" (in the sense of "the modifications are the same"), then you should use `fetchpatch` instead of copy-paste vendoring the patch here in Nixpkgs.
2. Put a descriptive reason why this patch is needed.
3. Is there any possibility this patch will be upstreamed? If so, comment with something like "remove this when next version arrives".

Reply:
I had problems with the remote patch until I found out I had to add `patchFlags = [ "-p0" ];`